addons/website_payment/models/payment_transaction.py

```diff
@@ -0,0 +1,61 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import _, fields, models
+
+
+class PaymentTransaction(models.Model):
+    _inherit = "payment.transaction"
+
+    is_donation = fields.Boolean(string="Is donation", help="Is the payment a donation")
+
+    def _finalize_post_processing(self):
+        for tx in self:
+            if tx.is_donation:
+                tx._send_donation_email()
+        super()._finalize_post_processing()
+
+    def _send_donation_email(self, notif=False, comment=None, recipient_email=None):
+        self.ensure_one()
+        tx_sudo = self.sudo()
```

Comment:
The method is already called with a 'sudoed' env. You should never need to use `sudo` from within a `payment.*` method.